semgrep-core/src/analyzing/Constant_propagation.ml

```diff
@@ -563,17 +562,25 @@ let propagate_basic lang prog =
                 attrs;
                 _;
               },
-              VarDef { vinit = Some { e = L literal; _ }; _ } )
+              VarDef { vinit = Some e; _ } )
           (* note that some languages such as Python do not have VarDef.
            * todo? should add those somewhere instead of in_lvalue detection?*)
             -> (
               match Hashtbl.find_opt stats (H.str_of_ident id, sid) with
               | Some stats ->
-                  if
-                    H.has_keyword_attr Const attrs
-                    || H.has_keyword_attr Final attrs
-                    || (!(stats.lvalue) = 1 && is_js env)
```

Comment:
@aryx do you think it's OK to remove `is_js env` here? Intuitively, it should be. But we handle so many languages and our const inference may not be perfect, that I'm a bit concerned we could start producing FPs in some places. Although our test suite seems to pass.

Reply:
Actually returntocorp/semgrep-rules#2358 is already an example where we are flagging something that we shouldn't, isn't it? Just because the a class attribute is assigned once _inside the class itself_, it doesn't mean it cannot be assigned from outside. And const-prop is meant to be a _must_ analysis.